protobuf-find-latest/src/main/java/io/apicurio/registry/examples/simple/protobuf/ProtobufFindLatestExample.java

```diff
@@ -0,0 +1,203 @@
+                                .setName("bb")
+                                .build())
+                        .build();
+
+                // Send/produce the message on the Kafka Producer
+                ProducerRecord<Object, AddressBook> producedRecord = new ProducerRecord<>(topicName, key, book);
+                producer.send(producedRecord);
+
+                Thread.sleep(100);
+            }
+            System.out.println("Messages successfully produced.");
+        } finally {
+            System.out.println("Closing the producer.");
+            producer.flush();
+            producer.close();
+        }
+
+        // Create the consumer
+        System.out.println("Creating the consumer.");
+        KafkaConsumer<Long, AddressBook> consumer = createKafkaConsumer();
+
+        // Subscribe to the topic
+        System.out.println("Subscribing to topic " + topicName);
+        consumer.subscribe(Collections.singletonList(topicName));
+
+        // Consume the 5 messages.
+        try {
+            int messageCount = 0;
+            System.out.println("Consuming (2) messages.");
+            while (messageCount < 2) {
+                final ConsumerRecords<Long, AddressBook> records = consumer.poll(Duration.ofSeconds(1));
+                messageCount += records.count();
+                if (records.count() == 0) {
+                    // Do nothing - no messages waiting.
+                    System.out.println("No messages waiting...");
+                } else records.forEach(record -> {
+                    AddressBook value = record.value();
+                    System.out.println("Consumed a message: People count in AddressBook " + value.getPeopleCount());
+                });
+            }
+        } finally {
+            consumer.close();
+        }
+
+        System.out.println("Done (success).");
+    }
+
+    /**
+     * Creates the Kafka producer.
+     */
+    private static Producer<Object, AddressBook> createKafkaProducer() {
+        Properties props = new Properties();
+
+        // Configure kafka settings
+        props.putIfAbsent(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, SERVERS);
+        props.putIfAbsent(ProducerConfig.CLIENT_ID_CONFIG, "Producer-" + TOPIC_NAME);
+        props.putIfAbsent(ProducerConfig.ACKS_CONFIG, "all");
+        props.putIfAbsent(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+        // Use the Apicurio Registry provided Kafka Serializer for Protobuf
+        props.putIfAbsent(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, ProtobufKafkaSerializer.class.getName());
+
+        // Configure Service Registry location
+        props.putIfAbsent(SerdeConfig.REGISTRY_URL, REGISTRY_URL);
+        props.putIfAbsent(SerdeConfig.EXPLICIT_ARTIFACT_GROUP_ID, "default");
+
+        // Find and use the latest artifact in the registry for the corresponding GroupId and ArtifactId
+        props.putIfAbsent(SerdeConfig.FIND_LATEST_ARTIFACT, Boolean.TRUE);
```

Comment:
You aren't setting AUTO_REGISTER here.  The default for that is `true` I think?  How do these two configs work together? E.g. what happens if I set `AUTO_REGISTRY=true` **and** `FIND_LATEST_ARTIFACT=true`?

Reply:
the default for `auto-register` is false, and the same for `find-latest`. How this configs work together is defined in the `DefaultSchemaResolver` implementation. For now it does: 
- 1. if `auto-register` is true try to create the artifact
- 2. if `find-latest` is true try to find the artifact by the resolved `ArtifactReference`
- 3. if the schema is available beforehand (avro and protobuf) try to find the artifact by content
- 4 fallback (jsonschema would be the only one to reach here) to find the artifact by `ArtifactReference`

I went with this approach to be slightly similar to the old serdes, where the find by content was the default. But we can change this whatever other way we think it makes more sense.